R/ggMarginal.R

```diff
@@ -397,71 +452,52 @@ ggMarginal <- function(p, data, x, y, type = c("density", "histogram", "boxplot"
                                     labels = ylabel)      
     }
   }
-
   # Create the vertical margin plot
   if (margins != "x") {
-    right <- marginPlot("y")
-    right <- addMainTheme(right, "y")
+    right <- marginPlot(margin = "y", type = type, xvar = xvar, yvar = yvar, 
+                      xparams = xparams, yparams = yparams, pb = pb, data = data,
+                      extraParams = extraParams)
+    right <- addMainTheme(marginal = right, margin = "y", p = p)
     right <- right +
       ggplot2::ylab(p$labels$x) +
-      getScale("y")
-  }
-
-  # Build a 2x2 grid for the scatterplot with marginal plots,
-  # with an extra row for the title
-  ncol <- 2
-  nrow <- 3
-  titleSize <- 1
-  rowSize <- 1
-  colSize <- 1
-
-  # Build the grid of grobs
-  empty <- grid::grid.rect(gp = grid::gpar(col = "transparent"), draw = FALSE)
-  if (!is.null(pb$plot$labels$title)) {
-    title <- grid::textGrob(
-      pb$plot$labels$title,
-      gp = grid::gpar(col = pb$plot$theme$plot.title$colour,
-                      fontsize = 16)
-    )
-    p$labels$title <- NULL
-  } else {
-    title <- empty
-    titleSize <- 0
+      getScale(margin = "y", type = type, pb = pb)
   }
   
-  
+  # Now add the marginal plots to the scatter plot
+  pGrob <- ggplot2::ggplotGrob(p)
+  suppressMessages({
   if (margins == "both") {
-    plots <- list(title, empty, top, empty, p, right)
-  } else if (margins == "x") {
-    plots <- list(title, empty, top, empty, p, empty)
-    colSize <- 0
-  } else if (margins == "y") {
-    plots <- list(title, empty, empty, empty, p, right)
-    rowSize <- 0
+        ggxtra_tmp <- addTopMargPlot(ggMargGrob = pGrob, top = top, 
+                                     size = size)
+        ggxtra_nottl <- addRightMargPlot(ggMargGrob = ggxtra_tmp, right = right, 
+                                         size = size)
+    } else if (margins == "x") {
+        ggxtra_tmp <- gtable::gtable_add_padding(x = pGrob, 
+                                                 grid::unit(c(0, .5, 0, 0), "lines"))
```

Comment:
The hard coding of .5 lines could be an issue if someone chooses huge font for their axes labs. I didn't think it would happen enough to warrant adding a param.